### PR TITLE
Fix Move not cancelling during turns.

### LINKED ISSUE #16976: Infantry do not always immediately cancel move activities
They sometimes continue for at least one more cell (and sometimes several!). I suspect this may be related to the L-move edge case.

Attached replay from @drogoganor provides a clear demonstration of the issue on `devtest-20190818`:

[OpenRA-Pathing-bug-demo.orarep.zip](https://github.com/OpenRA/OpenRA/files/3533766/OpenRA-Pathing-bug-demo.orarep.zip)

### PR DESCRIPTION
**Pull request by tovl**
Fixes #16976 

`Move` will not cancel during a turn. This problem is not exclusive to infantry, but is probably more annoying with infantry because it comes on top of the separate problem of infantry not choosing the closest subcell.

The root of the problem is that `MoveFirstPart` will keep adding new instances of `MoveFirstPart` into it's parents childactivity queue as long as the unit is making turns. Because `MovePart` is uninterruptible this makes `Move` uninterruptible while making a turn. A proper fix will probably entail moving all the queueing of `MovePart` into the parent activity and not letting the childactivities mess with that. However, that will be a major refactor of the `Move` code which will likely cause other regressions along the way.

The quick fix proposed here for next release/playtest is to clear the path upon cancelling (which was the case before the whole childactivity/cancellation refactor as well IIRC). I also removed the pointless `IsCancelling` clause from `MoveFirstPart`.

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Move/Move.cs [hunk 1]
```diff
@@ -259,6 +259,13 @@ public override bool Tick(Actor self)
 			return Pair.New(nextCell, subCell);
 		}
 
+		public override void Cancel(Actor self, bool keepQueue = false)
+		{
+			path.Clear();
```
Comment (pchote) [inline OpenRA.Mods.Common/Activities/Move/Move.cs]: `path` can be null, according to the NRE exception that I just found on the shellmap.

**Review by pchote — DISMISSED** [review]
LGTM after adding the null check

File: OpenRA.Mods.Common/Activities/Move/Move.cs (hunk 1 above)
Comment (tovl) [reply OpenRA.Mods.Common/Activities/Move/Move.cs]: Fixed

**Review by pchote — APPROVED** [review]

**Review by teinarss — APPROVED** [review]

**Comment by teinarss** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/65a75d1782313f365b0b575a96fde984a67c6868...3aa7489cc54db6b716902b626dc25090d9a61eed

**Pull request merged**